File: backend/src/main/java/com/adhamamr/passwordy/dto/AuthResponse.java
package com.adhamamr.passwordy.dto;

/**
 * Returned on successful login or token refresh. {@code token} is a short-lived access JWT;
 * {@code refreshToken} is the long-lived opaque token for {@code /api/auth/refresh}.
 *
 * <p>When the account has 2FA enabled, login returns {@code twoFactorRequired = true} with a
 * short-lived {@code twoFactorToken} and <b>no</b> access/refresh tokens — the client completes
 * login at {@code /api/auth/2fa/verify}.
 */
public record AuthResponse(
        String token,
        String refreshToken,
        String username,
        String email,
        String message,
        boolean twoFactorRequired,
        String twoFactorToken) {
}

File: frontend/app/src/main/java/com/adhamamr/passwordy/data/model/AuthResponse.kt
package com.adhamamr.passwordy.data.model

/**
 * Login / refresh response. [token] is the short-lived access JWT; [refreshToken] is the
 * long-lived opaque token. When [twoFactorRequired] is true, [token]/[refreshToken] are null and
 * [twoFactorToken] carries the short-lived challenge to complete at /api/auth/2fa/verify.
 */
data class AuthResponse(
    val token: String?,
    val refreshToken: String?,
    val username: String,
    val email: String,
    val message: String,
    val twoFactorRequired: Boolean = false,
    val twoFactorToken: String? = null
)
